trieste/models/gpflow/sampler.py

```diff
@@ -527,7 +527,7 @@ def _prepare_theta_posterior_in_design_space(self) -> tfp.distributions.Multivar
         and observation noise variance :math:`\sigma^2`.
         """
         dataset = self._model.get_internal_data()
-        phi = self._feature_functions(dataset.query_points)  # [n, m]
+        phi = self._feature_functions(tf.convert_to_tensor(dataset.query_points))  # [n, m]
```

Comment:
`Dataset` object should always have tensors, no?

Reply:
right, its `TensorType`, and although in practice we typically use tensors, it could be a numpy array

Reply:
Yes, I wasn't 100% sure what it could be, so I went for the safe option :-)